### ISSUE TITLE
Replace functions returning explicit `undefined` value

### ISSUE THREAD

**Issue by gildub** [trustification/trustify-ui#77]
Undefined means no value has been assigned where null means an empty value.
It's a good practice to return and propagate a null value instead of using `undefined` because it requires to handle the special type case of `undefined`.

**Comment by helio-frota** [comment]
I would add this practice as well:

before
```ts
if (num === undefined) {
  return "";
}
return String(num);
```

after
```ts
if (!num) {
  return "";
}
return String(num);
```

or more fancy

```ts
export const numStr = (num: number | undefined): string => String(num ?? "");

console.log(numStr(1));
console.log(numStr(0));
console.log(numStr(undefined));
```

[ts-playground](https://www.typescriptlang.org/play/?module=1#code/KYDwDg9gTgLgBAYwgOwM72QVwLYGUZRwC8cAFFtgFxwUBGwhAPnJsgCbABmAlssGwEpq6KLwDmxAHxx8o5GPI44AfmVwAROoEBuAFC6kaCABtgAOmMQFFWaQCMAnQZSoT5y9Zy2ADI72HXUwsrRTwCUlYOHj5BJyA)

**Comment by carlosthe19916** [comment]
I am not a fan of generic rules applied without reasoning. If there is a specific block of code where we consider we can achieve better readability, usability, performance, etc. then let's do it; let's open a PR and we can discuss what we are solving with that code change.

**Comment by helio-frota** [comment]
I think this is related with @gildub issue description to avoid runtime error https://github.com/reactwg/react-18/discussions/75
the idea is to delegate the check to the lint as trustify-ui is using react 18.

**Comment by helio-frota** [comment]
For this we would need to enable the lint https://github.com/trustification/trustify-ui/blob/main/.github/workflows/ci-actions.yaml#L29
and add the suggested rules (I have no idea if there are other rules for this-react-related) on https://github.com/trustification/trustify-ui/blob/main/.eslintrc.cjs#L40

```
    "consistent-return": "error", ( or warn ?)
    "no-unused-expressions": "error", ( or warn ?)
```

**Comment by carlosthe19916** [comment]
Cleaning some issues : @gildub is this issue still relevant?

**Comment by gildub** [comment]
@carlosthe19916, yes absolutely. There are few places where we need to remove assigning or returning `undefined`.

**Comment by carlosthe19916** [comment]
An idea came to my mind, we will use Biome already for scanning the code and apply Biome Rules. E.g. https://github.com/trustification/trustify-ui/pull/409

The PR https://github.com/trustification/trustify-ui/pull/409 didn't highlight anything related to `undefined` so in order to make the issue more concrete we could perhaps find a Biome rule that enforces it. I don't know if such thing even exists though.

**Comment by carlosthe19916** [comment]
Closing issue as so far no issue was identified by the fact that we use undefined in the current code. When specific issues are identified we can reopen this issue
